webapp/packages/plugin-sql-editor/src/SqlResultTabs/OutputLogs/OutputLogsService.ts

```diff
@@ -26,9 +27,24 @@ export class OutputLogsService {
     return this.userDataService.getUserData(OUTPUT_LOGS_KEY, getOutputLogsDefaultSettings);
   }
 
-  constructor(private readonly sqlDataSourceService: SqlDataSourceService, private readonly userDataService: UserDataService) {}
+  constructor(
+    private readonly sqlDataSourceService: SqlDataSourceService,
+    private readonly userDataService: UserDataService,
+    private readonly outputLogsResource: OutputLogsResource,
+  ) {
+    this.clearOutputLogs = this.clearOutputLogs.bind(this);
 
-  async showOutputLogs(editorState: ISqlEditorTabState): Promise<void> {
+    makeObservable(this, {
+      clearOutputLogs: action.bound,
+      settings: computed,
+      getOutputLogs: action.bound,
+      toggleWrapMode: action.bound,
+      showOutputLogs: action.bound,
+      removeOutputLogsTab: action.bound,
+    });
```

Comment:
I'm not a fan of binding methods every time. I think it's better to bind only those methods that really need it

Moreover, sometimes you don't need `action`, for example: `this.settings.wrapMode = !this.settings.wrapMode`. This is an atomic action, so there is nothing to batch (the action will have no effect there)

`getOutputLogs` can't be an action, this will break observability of data, action only for methods that mutate several observable states, it's not supposed to be used with getters